build-config/swe_checkstyle.xml

```diff
@@ -1,140 +1,140 @@
-        <module name="MagicNumber"/>
-        <module name="MissingSwitchDefault"/>
-        <module name="SimplifyBooleanExpression"/>
-        <module name="SimplifyBooleanReturn"/>
-        <module name="CovariantEquals"/>
-        <module name="DefaultComesLast"/>
-        <module name="ModifiedControlVariable"/>
-        <module name="StringLiteralEquality"/>
-        <module name="FinalClass"/>
-        <module name="HideUtilityClassConstructor"/>
-        <module name="InterfaceIsType">
-                <property name="severity" value="warning"/>
-        </module>
-        <module name="VisibilityModifier">
-                <property name="packageAllowed" value="true"/>
-                <property name="protectedAllowed" value="true"/>
-        </module>
-        <module name="ArrayTypeStyle"/>
-        <module name="TodoComment">
-                <property name="severity" value="ignore"/>
-        </module>
-        <module name="UpperEll"/>
-        <module name="JavadocMethod">
-                <property name="logLoadErrors" value="true"/>
-                <property name="suppressLoadErrors" value="true"/>
-        </module>
-        <module name="JavadocStyle">
-                <property name="scope" value="protected"/>
-        </module>
-        <module name="JavadocType"/>
-        <module name="JavadocVariable">
-                <property name="scope" value="protected"/>
-        </module>
-        <module name="FileContentsHolder"/>
-        <module name="MultipleVariableDeclarations" >
-                <property name="severity" value="warning"/>
-        </module>
-        <module name="ArrayTrailingComma">
-                <property name="severity" value="warning"/>
-        </module>
-    </module>
-    <module name="NewlineAtEndOfFile"/>
-    <!--<module name="Translation">
-        <property name="severity" value="info"/>
-    </module> Translations are lagging behind anyways-->
-    <!-- Ignore generated code -->
-    <module name="SuppressionCommentFilter">
-        <property name="offCommentFormat" value="This file was auto-generated from WSDL"/>
-    </module>
-    <module name="SuppressWithNearbyCommentFilter">
-        <property name="commentFormat" value="CS IGNORE (\w+) FOR NEXT (\d+) LINES\. REASON\: \w+"/>
-        <property name="checkFormat" value="$1"/>
-        <property name="influenceFormat" value="$2"/>
-    </module>
-</module>
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+    This configuration file was written by the eclipse-cs plugin configuration editor
+-->
+<!--
+    Checkstyle-Configuration: SWE Checkstyle checks
+    Description: This is the SW Environment Checkstyle template
+-->
+<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.2//EN" "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">
+<module name="Checker">
+    <property name="severity" value="error"/>
+    <module name="FileLength">
+        <!-- TODO: Need refactoring -->
+        <property name="max" value="2600"/>
```

Comment:
Is there some way you could try to refactor the issue instead of patching the rules? 2500 is already a bit too big of a file :)

Reply:
One of the options is to remove deprecated code 

* deprecated constructor 112 lines
* deprecated schedule (2 methods) 26 lines
* deprecated ParametersAction 13 lines
* deprecated isNoNameAndEmailParameters (2 methds) 30 lines

(230 lines) 10% of the code
deprecated setReadableMessage (32 lines)
deprecated data members 18 lines

Reply:
Too worried about backwards compatibility to encourage removing deprecated data and signatures.
No static methods that could move to a utility class or common code that could be refactored into utility methods?

Reply:
The bulk of the code is setters and logic. One viable option is to move the inner classes out of the implementation and possibly rename them to have more meaningful names.

class DescriptorImpl out of the implementation. (300 lines)
class RunningJobs out of the implementation (188 lines)

and other "funny" option is to move the deprecated methods to a parent class

Reply:
I implemented DescriptorImpl as top level class in #413

Reply:
No `DescriptorImpl` should always be inside the class it is describing, even though it is possible to do by overriding the `getDescriptor` method, it's not common practice, so we should try to avoid that.

`RunningJobs` could be moved out, that would be good for some cleaning up.
Moving deprecated methods to a parent class also sounds like a good idea, would be really nice to get those cleaned up :)